File: TiendaAPI/Entities/Producto_Carro.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace TiendaAPI.Entities
{
    public class Producto_Carro
    {   

        [Key]
        public int idProducto_Carro { get; set; }
        
        public int idCarro { get; set; }
        
        public int idProducto { get; set; }
        public int cantidad { get; set; }
        [ForeignKey("idCarro")]
        public virtual Carro Carro { get; set; }

        [ForeignKey("idProducto")]
        public virtual Producto Producto { get; set; }
        

    }
}

File: TiendaApp/src/app/models/producto-carro.ts
import { Producto } from './producto';
import { Carro } from './carro';
export class ProductoCarro {

    idProducto_Carro:number;
    idCarro:number;
    idProducto:number;
    cantidad:number;
    producto:Producto

    constructor(idProducto_Carro:number,idCarro:number,idProducto:number,cantidad:number,Producto:Producto){
        this.idProducto_Carro = idProducto_Carro;
        this.idCarro = idCarro;
        this.idProducto = idProducto;
        this.cantidad = cantidad;
        this.producto = Producto
    }    

}
